[v1.17] daemon: deprecate no-op `--k8s-watcher-endpoint-selector` flag

tklauser opened the pull request:
The `--k8s-watcher-endpoint-selector` flag no longer has any effect since Cilium v1.14. Namely, in #23977 endpoint watchers were replaced by `Resource[*Endpoint]` and thus this configuration is no longer applicable.

```release-note
daemon: deprecate no-op --k8s-watcher-endpoint-selector flag
```

joestringer reviewed (approved):
Needs a rebase but LGTM :+1:

tklauser committed:
daemon: deprecate no-op --k8s-watcher-endpoint-selector flag

The --k8s-watcher-endpoint-selector flag no longer has any effect since
Cilium v1.14. Namely, in commit 82a728aa1780 ("agent, operator,
clustermesh-apiserver: use Resource[*Endpoints]") endpoint watchers were
replaced by Resource[*Endpoint] and thus this configuration is no longer
applicable.

Signed-off-by: Tobias Klauser <tobias@cilium.io>
